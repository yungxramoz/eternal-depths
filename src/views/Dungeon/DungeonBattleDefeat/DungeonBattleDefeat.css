.defeat-container {
  height: 100%;
  display: flex;
  align-items: center;
  flex-direction: column;
  justify-content: center;
  gap: 1em;
  padding: 1em;
  text-align: center;
}
