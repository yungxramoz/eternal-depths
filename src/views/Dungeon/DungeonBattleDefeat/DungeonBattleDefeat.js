import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import RpgButton from '../../../components/atoms/RpgButton/RpgButton'
import RpgIcon from '../../../components/atoms/RpgIcon/RpgIcon'
import { gameReset } from '../../../store/game/gameSlice'
import './DungeonBattleDefeat.css'

const DungeonBattleDefeat = () => {
  const dispatch = useDispatch()
  const stage = useSelector((state) => state.game.stage)

  const handleRetry = () => {
    dispatch(gameReset())
  }

  return (
    <div className="defeat-container">
      <h2>You Died!</h2>
      <h3>Stage {stage}</h3>
      <RpgIcon icon="skull" />
      <p>
        In the heat of a fierce battle, you met your end. Your name, once
        whispered in the winds, will now fade into silence of oblivion.
      </p>
      <RpgButton text="End" onClick={handleRetry} />
    </div>
  )
}

export default DungeonBattleDefeat
