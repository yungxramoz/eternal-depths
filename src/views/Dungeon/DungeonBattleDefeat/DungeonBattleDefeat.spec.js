import { configureStore } from '@reduxjs/toolkit'
import { fireEvent, render, screen } from '@testing-library/react'
import React from 'react'
import { Provider } from 'react-redux'
import gameReducer from '../../../store/game/gameSlice'
import DungeonBattleDefeat from './DungeonBattleDefeat'

describe('DungeonBattleDefeat', () => {
  let store
  beforeEach(() => {
    store = configureStore({ reducer: { game: gameReducer } })
  })

  it('renders correctly', () => {
    render(
      <Provider store={store}>
        <DungeonBattleDefeat />
      </Provider>,
    )
    expect(document.body).toMatchSnapshot('DungeonBattleDefeat')
  })

  it('dispatches gameReset when button is clicked', () => {
    const dispatchSpy = jest.spyOn(store, 'dispatch')
    render(
      <Provider store={store}>
        <DungeonBattleDefeat />
      </Provider>,
    )
    fireEvent.click(screen.getByText('End'))
    expect(dispatchSpy).toHaveBeenCalledWith(
      expect.objectContaining({ type: 'game/gameReset' }),
    )
  })
})
