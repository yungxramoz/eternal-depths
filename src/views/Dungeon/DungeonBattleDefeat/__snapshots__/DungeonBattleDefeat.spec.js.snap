// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DungeonBattleDefeat renders correctly: DungeonBattleDefeat 1`] = `
<body>
  <div>
    <div
      class="defeat-container"
    >
      <h2>
        You Died!
      </h2>
      <h3>
        Stage 
        0
      </h3>
      <div
        class="rpgui-icon skull "
      />
      <p>
        In the heat of a fierce battle, you met your end. Your name, once whispered in the winds, will now fade into silence of oblivion.
      </p>
      <button
        class="rpgui-button btn"
      >
        End
      </button>
    </div>
  </div>
</body>
`;
